cl_manycore/libraries/bsg_manycore_coordinate.h

```diff
@@ -19,7 +23,7 @@ typedef enum __hb_mc_coordinate_limit_t {
         HB_MC_COORDINATE_MAX = 32
 } hb_mc_coordinate_limit_t;
 
-typedef int32_t hb_mc_idx_t;
+typedef uint32_t hb_mc_idx_t;
```

Comment:
Is there any reason you changed this to a uint32_t?

Reply:
Well first of all I figured why have it as int32 since a negative index doens't make sense, and secondly int the previous API before the refactor all indexes were declared as uint32, so I figured it would be safer to match that. Is there any reason why having it has int32 would make more sense?

Reply:
I'm not sure why it was a signed int to begin with. @drichmond why was this made signed?

Reply:
Because I thought that we could return some codes using negative integers.

Reply:
Why would we return error codes where we should have returned a coordinate value? If a function can return an error, it should be returning an int.

Reply:
Relax, I didn't say it was a well thought out idea